### PR TITLE
Fix #1002: Adding new command to list details of the VCS branches / tags of an application

### LINKED ISSUE #1002: CLI-764: [New  Command] Get list of used and un-used tags and branches
We can get list of all available tags and branches from `acli api:applications:vcs-info` but we don't have any info to know which one is in use and which one is not. For a application with dev, stage and prod - checking via  acquia cloud is quick but if there are many environments available for an application, it will be difficult.

### PR DESCRIPTION
**Pull request by joshirohit100**
Fix #1002

Old pull request which was closed - https://github.com/acquia/cli/pull/1003

Screenshot 2023-01-14 at 1 21 58 PM

### CODE REVIEW

**Comment by danepowell** [comment]
I like this, let's clean it up and get it merged.

File: src/Command/App/AppBranchesTagsInfo.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Class AppBranchesTagsInfo.
+ */
+class AppBranchesTagsInfo extends CommandBase {
+
+  protected static $defaultName = 'app:get-branches-tags';
```
Comment (danepowell) [inline src/Command/App/AppBranchesTagsInfo.php:17]: Let's say "list" instead of "get" and use colons instead of dashes. Also "branches-tags" is a little awkward.

Maybe `app:list:vcs`? `app:vcs:refs`? `app:list:deployments`?

@anavarre opinions?

Comment (anavarre) [reply src/Command/App/AppBranchesTagsInfo.php:17]: If we're looking for purity, the closest to that is `app:vcs:refs`. In plain english it means I want to display the version control system references that are deployed or not for my application, which is exactly what the command is doing.

Comment (danepowell) [reply src/Command/App/AppBranchesTagsInfo.php:17]: I like `app:vcs:refs`. @joshirohit100 would this be intuitive to you as a developer?

Comment (joshirohit100) [reply src/Command/App/AppBranchesTagsInfo.php:17]: To me as dev something like `app:vcs:info` is more intuitive.

Comment (joshirohit100) [reply src/Command/App/AppBranchesTagsInfo.php]: Updated the PR with the command name `app:vcs:info`. Added two more tests.

Comment (anavarre) [reply src/Command/App/AppBranchesTagsInfo.php]: I can get behind `app:vcs:info`

**Comment by anavarre** [comment]
I have been playing with the command and it does the job.

```
❯ ./acli.phar app:vcs:info REDACTED
+------------- Status of Branches and Tags of the Application --------------+
| Branch / Tag Name                       | Deployed | Deployed Environment |
+-----------------------------------------+----------+----------------------+
| pipelines-build-master                  | Yes      | Stage                |
| tags/2020-09-28                         | Yes      | Prod                 |
| blt                                     | No       | None                 |
| composer                                | No       | None                 |
| dane                                    | No       | None                 |
| drush-codestudio-build                  | No       | None                 |
| from-ide                                | No       | None                 |
| integration                             | No       | None                 |
| master                                  | No       | None                 |
| master-codestudio-build                 | No       | None                 |
| php-codestudio-build                    | No       | None                 |
| pipelines-build-GH1                     | No       | None                 |
| pipelines-build-GH3                     | No       | None                 |
| pipelines-build-GH5                     | No       | None                 |
| pipelines-build-GH7                     | No       | None                 |
| pipelines-build-GH9                     | No       | None                 |
| pipelines-build-GL-1                    | No       | None                 |
| pipelines-build-ads-integration         | No       | None                 |
| pipelines-build-codesnippet             | No       | None                 |
| pipelines-build-core-8.6.10             | No       | None                 |
| pipelines-build-include-file            | No       | None                 |
| pipelines-build-linkit                  | No       | None                 |
| pipelines-build-local-dev               | No       | None                 |
| pipelines-build-missing-dotfiles        | No       | None                 |
| pipelines-build-missing-files           | No       | None                 |
| pipelines-build-modules                 | No       | None                 |
| pipelines-build-mysql57                 | No       | None                 |
| pipelines-build-patch-codesnippet       | No       | None                 |
| pipelines-build-pipelines               | No       | None                 |
| pipelines-build-pipelines-73            | No       | None                 |
| pipelines-build-pipelines-74            | No       | None                 |
| pipelines-build-pipelines-behat         | No       | None                 |
| pipelines-build-pipelines-build-master  | No       | None                 |
| pipelines-build-pr-1                    | No       | None                 |
| pipelines-build-pr-10                   | No       | None                 |
| pipelines-build-pr-11                   | No       | None                 |
| pipelines-build-pr-12                   | No       | None                 |
| pipelines-build-pr-13                   | No       | None                 |
| pipelines-build-pr-14                   | No       | None                 |
| pipelines-build-pr-18                   | No       | None                 |
| pipelines-build-pr-19                   | No       | None                 |
| pipelines-build-pr-2                    | No       | None                 |
| pipelines-build-pr-20                   | No       | None                 |
| pipelines-build-pr-21                   | No       | None                 |
| pipelines-build-pr-3                    | No       | None                 |
| pipelines-build-pr-4                    | No       | None                 |
| pipelines-build-pr-5                    | No       | None                 |
| pipelines-build-pr-6                    | No       | None                 |
| pipelines-build-pr-7                    | No       | None                 |
| pipelines-build-pr-8                    | No       | None                 |
| pipelines-build-pr-9                    | No       | None                 |
| pipelines-build-redirect                | No       | None                 |
| pipelines-build-require-codesnippet-lib | No       | None                 |
| pipelines-build-themes                  | No       | None                 |
| pipelines-build-update-drupal           | No       | None                 |
| pipelines-build-url-embed               | No       | None                 |
| pipelines-build-vcs                     | No       | None                 |
| theiagit                                | No       | None                 |
| tags/2015-12-17                         | No       | None                 |
| tags/2015-12-17.0                       | No       | None                 |
| tags/2015-12-18                         | No       | None                 |
| tags/2015-12-21                         | No       | None                 |
| tags/2016-01-07                         | No       | None                 |
| tags/2016-02-04                         | No       | None                 |
| tags/2016-02-20                         | No       | None                 |
| tags/2016-02-20.0                       | No       | None                 |
| tags/2016-02-25                         | No       | None                 |
| tags/2016-03-03                         | No       | None                 |
| tags/2016-03-04                         | No       | None                 |
| tags/2016-03-30                         | No       | None                 |
| tags/2016-06-07                         | No       | None                 |
| tags/2016-06-07.0                       | No       | None                 |
| tags/2016-06-07.1                       | No       | None                 |
| tags/2016-06-08                         | No       | None                 |
| tags/2016-06-09                         | No       | None                 |
| tags/2016-06-15                         | No       | None                 |
| tags/2016-06-15.0                       | No       | None                 |
| tags/2016-12-15                         | No       | None                 |
| tags/2016-12-22                         | No       | None                 |
| tags/2017-03-20                         | No       | None                 |
| tags/2017-03-21                         | No       | None                 |
| tags/2017-03-21.0                       | No       | None                 |
| tags/2017-03-21.1                       | No       | None                 |
| tags/2017-03-21.2                       | No       | None                 |
| tags/2017-04-15                         | No       | None                 |
| tags/2017-04-15.0                       | No       | None                 |
| tags/2017-04-15.1                       | No       | None                 |
| tags/2017-06-16                         | No       | None                 |
| tags/2017-06-16.0                       | No       | None                 |
| tags/2017-06-16.1                       | No       | None                 |
| tags/2017-06-20                         | No       | None                 |
| tags/2017-06-20.0                       | No       | None                 |
| tags/2017-06-20.1                       | No       | None                 |
| tags/2017-06-20.2                       | No       | None                 |
| tags/2017-11-08                         | No       | None                 |
| tags/2017-11-11                         | No       | None                 |
| tags/2017-11-12                         | No       | None                 |
| tags/2018-03-28                         | No       | None                 |
| tags/2018-03-28.0                       | No       | None                 |
| tags/2019-02-19                         | No       | None                 |
| tags/2019-02-20                         | No       | None                 |
| tags/2019-02-23                         | No       | None                 |
| tags/2019-02-23.0                       | No       | None                 |
| tags/2019-07-20                         | No       | None                 |
| tags/2019-07-20.0                       | No       | None                 |
| tags/2020-03-14                         | No       | None                 |
| tags/2020-03-14.0                       | No       | None                 |
| tags/2020-03-14.1                       | No       | None                 |
| tags/2020-03-14.2                       | No       | None                 |
| tags/2020-04-06                         | No       | None                 |
| tags/2020-10-19                         | No       | None                 |
| tags/WELCOME                            | No       | None                 |
| tags/pre-import-2015-12-10              | No       | None                 |
| tags/pre-import-2015-12-10.0            | No       | None                 |
| tags/pre-import-2015-12-10.1            | No       | None                 |
| tags/pre-import-2015-12-11              | No       | None                 |
| tags/pre-import-2020-03-10              | No       | None                 |
| tags/pre-import-2022-05-11              | No       | None                 |
| tags/progress                           | No       | None                 |
| tags/test-ads                           | No       | None                 |
| tags/testing-ads-deploy                 | No       | None                 |
+-----------------------------------------+----------+----------------------+
```

Should we support a `--deployed` param so you can only return what you truly care about and nothing else? That is, refs that are deployed?

**Comment by danepowell** [comment]
`--deployed` might be nice... I'm going to clean this up and merge as-is, we can follow up with more additions.

**Review by danepowell — APPROVED** [review]
